etna/search/views.py

```diff
@@ -526,45 +526,20 @@ def get_selected_filters(self, form: Form) -> Dict[str, List[Tuple[str, str]]]:
         if filter_keyword := form.cleaned_data.get("filter_keyword"):
             return_value.update({"filter_keyword": [(filter_keyword, filter_keyword)]})
 
-        # get form error messages
-        if error_dict := json.loads(form.errors.as_json()):
-            for dict_values in error_dict.values():
-                for item in dict_values:
-                    form_error_messages.append(item["message"])
-
-        if opening_start_date := form.cleaned_data.get("opening_start_date"):
-            # if both dates have valid values but invalid when together
-            if (
-                CUSTOM_ERROR_MESSAGES.get("invalid_date_range")
-                not in form_error_messages
-            ):
-                return_value.update(
-                    {
-                        "opening_start_date": [
-                            (
-                                opening_start_date,
-                                opening_start_date.strftime(
```

Comment:
@TNA-Allan Where does this validation happen now? It should really belong to the form, but I can't see it there at a glance.

Reply:
@ababic  Its not a validation. Its where the filter labels are applied. 
see 
`# prepare filter labels for date fields
        for date_item in [
            ("opening_start_date", "Record Opening from: %d-%m-%Y"),
            ("opening_end_date", "Record Opening to:  %d-%m-%Y"),
            ("created_start_date", "Date from: %d-%m-%Y"),
            ("created_end_date", "Date to: %d-%m-%Y"),
        ]:`

Reply:
Oh, I see. You've removed this and decided to do it in the template instead then? That feels very backwards to me. You should revert the change and have the labels be prepared in the right format for the template to pick up.

Reply:
@TNA-Allan just please go and look how this method is used and what those values are used for.  I realise validation is not happening here, but you're still removing code that should be here and using unnecessary conditionals in the template to render the 'applied filters' bit. If the purpose of this method is to prepare values for the front end, then that's what is should be doing.

Reply:
@ababic 
I've checked the conditions for date error message prevents showing the selected date label on error. This is restored the code for this. With this the refactored condition for date labels is reverted, now the new date[ labels for created dates](https://github.com/nationalarchives/ds-wagtail/blob/ad810061e2d07b47510432f6e43272e8e8187515/etna/search/views.py#L539-L572) are added the same way as record opening dates.